frontend/modules/presentation/presentation.redux.ts

```diff
@@ -0,0 +1,83 @@
+/**
+ *  Copyright (C) 2020 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import { createActions, createReducer } from 'reduxsauce';
+
+const generateCode = () => {
```

Comment:
I think this should be done on the backend..

There should be an endpoint to start a presentation which returns this code. My thinking behind this:

- To avoid duplicated presenter code (especially with embedded viewer users we want to ensure this is controlled on our own backend)
- We should be checking if the user calling the endpoint to update the viewpoint is the presenter or not, so we need to know who created the presentation in the first place
- To be able to validate code (Is there anyway for audience to validate? what happens if they entered the wrong code?)
- it'll be built on and the database will have to store some info about the session, and people who are connecting late will need to have a snapshot of what is currently happening to catch up to